Core/Controllers/ReportsController.cs

```diff
@@ -44,6 +48,8 @@ public IActionResult SubmitReport([FromBody] ReportSubmission reportSubmission)
 
             if (currentUser.ID == targetEntry.Author.ID)
                 return BadRequest("Reporting own posts is not allowed.");
+            if (targetEntry.Reports.Any(r => r.ReportAuthor.ID == currentUser.ID))
+                return BadRequest("A report on this entry by the current user already exists.");
```

Comment:
I'd make the message more personal; "You already reported this entry."

Reply:
The user won't ever see this message though, unless the API request is manually forged.

Reply:
Oh I was under the impression this were returned to the user. Nothing to add then :)